### DISCUSSION TITLE
Maester enterprise app with delegate access

### DISCUSSION THREAD

**Discussion by pesip** [maester365/maester#452] [Show and tell]
## I need advice on how to create an Enterprise app to run all the tests.

***I am frequently using the module in interactive mode.***  If I have Global Administrator role, all available tests are executed. As seen in the following image:

![image](https://github.com/user-attachments/assets/0f8bf7ae-954c-4abe-8b35-a2f5504b3979)

***Sometimes this approach doesn't work, and it's necessary to create an Enterprise app with the api permissions like delegated permissions*** described in the documentation. As seen in the following image:

![image](https://github.com/user-attachments/assets/2fee30ca-f7c1-44a8-ad94-bfac6281d143)

 ***In this case, only 76 tests are executed.***  As seen in the following image:

![image](https://github.com/user-attachments/assets/f932723b-bf1d-470f-b187-c4bc714512a2)

**I need advice on how to create an Enterprise app to run all the tests.**

**Comment by pesip** [comment]
***These categories are missed:*** 

Authentication Method - FIDO2 security key
Authentication Method - General Settings
Authentication Method - Microsoft Authenticator
Authentication Method - Temporary Access Pass
Authentication Method - Voice call
Conditional Access WhatIf
Consent Framework - Admin Consent Request
Default Authorization Settings
Default Settings - Classification and M365 Groups
Default Settings - Consent Policy Settings
Default Settings - Password Rule Settings
Entra Recommendations

**Comment by soulemike** [comment]
Hi @pesip, did you setup the optional permissions, it looks like you may be missing `RoleEligibilitySchedule.ReadWrite.Directory`, then you'd also need to give the service principal access to an EXO and Azure role. Once those permissions are set you'd need to also call `Connect-Maester -Service All`. 

https://maester.dev/docs/monitoring/azure-container-app-job#optional-grant-permissions-to-exchange-online
https://maester.dev/docs/installation#connecting-to-azure-exchange-and-other-services

**Reply by pesip** [reply]
Hi, thanks for your response. I’m dealing with an issue where tests are missing when running between the Enterprise app and an interactive environment.

When I use GitHub Actions, everything works fine. Unfortunately, I can't use GitHub for all situations.

**Comment by magnusjak** [comment]
Hi,

I have an application that I use to run Maester through an Azure AutomationAccount. The application does other things as well, so I believe I have more access here than Maester needs, but this is working for me:

![image](https://github.com/user-attachments/assets/84dcd548-ee9c-4ba6-a99c-f20120ded2ae)

I use this app to generate tokens or directly use its certificate which I in turn use to connect to the different services needed (Connect-MgGraph, Connect-AzAccount, Connect-ExchangeOnline and Connect-IPPSSession), before using 'Connect-Maester -SkipGraphConnect'.

Not sure if this scenario is similar to yours though.
